driver-core/src/main/com/mongodb/internal/logging/StructuredLogMessage.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.logging;
+
+import com.mongodb.connection.ClusterId;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * <p>This class is not part of the public API and may be removed or changed at any time</p>
+ */
+public class StructuredLogMessage {
```

Comment:
It is helpful to have `@Nullable` annotations on fields, methods and formal parameters where applicable, instead of analyzing the code each time we try to use internal API.

The same is applicable to `StructuredLogger`.

Reply:
I made the package a NonNullApi and added Nullable as needed.